library/asb_set_state.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/python
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: asb_set_state
+short_description: Sets state in a ConfigMap
+description:
+     - Provides a secure method for allowing Service Bundle developers
+       to set state during a Service Bundle action.
+       Provided key, value pair is stored in a ConfigMap for later retrieval.
+author:
+    - "Red Hat, Inc."
+options:
+    key:
+        description:
+            - Unique key
+        required: true
+    value:
+        description:
+            - Literal value
+        required: true
+ env:
+        # Set via the downward API on the APB Pod
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+'''
+
+EXAMPLES = '''
+- name: Set some example state
+  asb_set_state:
+    key: "example-key"
+    value: "some-value"
+'''
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from kubernetes import client, config
+    from kubernetes.client.rest import ApiException
+    config.load_kube_config()
+    api = client.CoreV1Api()
```

Comment:
This should probably happen in the `main` function rather than on import, you could either then pass the `api` object through to the functions, or another common pattern is to make a class that holds that state